backend/app/services/posting_service.rb

```diff
@@ -65,14 +65,71 @@ def prefill(user:)
         applicant = Applicant.find_by(utorid: utorid)
         return { utorid: utorid } unless applicant
 
-        {
+        # This is the basic data we can get if the applicant already exists
+        data = {
             utorid: utorid,
             student_number: applicant.student_number,
             first_name: applicant.first_name,
             last_name: applicant.last_name,
             email: applicant.email,
             phone: applicant.phone
         }
+
+        existing_application = Application.find_by(posting: @posting)
+        if existing_application
+            application_service =
+                ApplicationService.new application: existing_application
+            data.merge! application_service.prefilled_data
+        else
+            # Find out if they have previous contracts in the database.
+            # This will tell us information about whether the applicant has been previously employed
+            offer_history =
+                Assignment.joins(:applicant, :offers, :position).where(
+                    applicant: applicant, "offers.status": 'accepted'
+                ).order("offers.position_start_date": :ASC).pluck(
+                    :"positions.position_code",
+                    :"offers.hours",
+                    :"offers.position_start_date",
+                    :"offers.position_end_date"
+                )
+            unless offer_history.blank?
```

Comment:
What is the difference between `blank?` and `empty?` I tried looking it up, but it was kind of confusing, do I understand correctly that for arrays these would be equivalent?

Reply:
`blank?` is more general than `empty?`. For example, `nil` has `blank?` but not `empty?`